tensorflow_datasets/scripts/cleanup/delete_old_versions.py

```diff
@@ -0,0 +1,111 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""Script to detect all the versions of the various datasets and delete the old/non-existing ones
```

Comment:
@Conchylicultor Could you help me out as to where i should put the put? in the script itself or the `tfds.testing` folder?

Reply:
In `scripts/cleanup/delete_old_datasets_test.py`

Reply:
Then test with `pytest tensorflow_datasets/scripts/cleanup/delete_old_datasets_test.py`

Reply:
@Conchylicultor I wrote the script for the test following your base template but after running it with pytest it gives a `FileNotFoundError` at the `tmp_path.touch()`.
The complete log:
```
================================================= test session starts =================================================
platform win32 -- Python 3.7.4, pytest-6.0.1, py-1.9.0, pluggy-0.13.1
rootdir: F:\tfds\datasets
collected 1 item

F:\tfds\datasets\tensorflow_datasets\scripts\cleanup\delete_old_versions_test.py F                               [100%]

====================================================== FAILURES =======================================================
_________________________________________________ test_delete_script __________________________________________________

tmp_path = WindowsPath('C:/Users/nikhil/AppData/Local/Temp/pytest-of-nikhil/pytest-1/test_delete_script0')

    def test_delete_script(tmp_path: pathlib.Path):
      for dataset_dir in [
        'dataset/config/1.0.0',
        'my_dataset/1.0.0',
        'my_dataset/1.2.0',
        'my_dataset/1.3.0',
        'my_other_dataset/config/1.2.0',
        'my_other_dataset/config/1.3.0',
        'my_other_dataset/other_config/1.2.0',
        'my_other_dataset/other_config/1.3.0',
        'old_dataset',
      ]:
>       tmp_path.joinpath(dataset_dir).touch()

F:\tfds\datasets\tensorflow_datasets\scripts\cleanup\delete_old_versions_test.py:33:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
f:\python\lib\pathlib.py:1248: in touch
    fd = self._raw_open(flags, mode)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = WindowsPath('C:/Users/nikhil/AppData/Local/Temp/pytest-of-nikhil/pytest-1/test_delete_script0/dataset/config/1.0.0')
flags = 257, mode = 438

    def _raw_open(self, flags, mode=0o777):
        """
        Open the file pointed by this path and return a file descriptor,
        as os.open() does.
        """
        if self._closed:
            self._raise_closed()
>       return self._accessor.open(self, flags, mode)
E       FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\nikhil\\AppData\\Local\\Temp\\pytest-of-nikhil\\pytest-1\\test_delete_script0\\dataset\\config\\1.0.0'

f:\python\lib\pathlib.py:1055: FileNotFoundError
================================================== warnings summary ===================================================
f:\python\lib\site-packages\win32\lib\pywintypes.py:2
  f:\python\lib\site-packages\win32\lib\pywintypes.py:2: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp, sys, os

f:\python\lib\site-packages\absl\testing\parameterized.py:200
  f:\python\lib\site-packages\absl\testing\parameterized.py:200: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    return (isinstance(obj, collections.Iterable) and

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=============================================== short test summary info ===============================================
FAILED F:\tfds\datasets\tensorflow_datasets\scripts\cleanup\delete_old_versions_test.py::test_delete_script - FileNot...
============================================ 1 failed, 2 warnings in 3.44s ============================================

```

What do you think is causing this?

Reply:
Please commit the file

Reply:
@vijayphoenix I have commited the changes done uptill now

Reply:
I think this should be `tmp_path.joinpath(dataset_dir, 'dataset_info.json').touch()`

Reply:
@Conchylicultor I tried that but instead of 
`FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\nikhil\\AppData\\Local\\Temp\\pytest-of-nikhil\\pytest-1\\test_delete_script0\\dataset\\config\\1.0.0'`
it is now simply giving 
` FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\nikhil\\AppData\\Local\\Temp\\pytest-of-nikhil\\pytest-3\\test_delete_script0\\dataset\\config\\1.0.0\\dataset_info.json'`
Everything else is the same as before. Do you have any ideas? @vijayphoenix

Reply:
@Conchylicultor The problem seems to be with the way `pathlib.Path.touch()` works. In the case of the first location in the path that it can't find it creates a file instead of a directory.
In the case of 
`FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\nikhil\\AppData\\Local\\Temp\\pytest-of-nikhil\\pytest-1\\test_delete_script0\\dataset\\config\\1.0.0'`
It creares a file by the ame of `test_delete_script0` instead of a directory and therefore cannot proceed further with the path

Reply:
@Conchylicultor The test successfully passes by using `tmp_path.join(data_dir).mkdir(parents=True)` and using `os.path.normpath()` before asserting